fe/fe-core/src/main/java/com/starrocks/sql/PlannerProfile.java

```diff
@@ -0,0 +1,133 @@
+// This file is made available under Elastic License 2.0.
+
+package com.starrocks.sql;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.common.util.RuntimeProfile;
+import com.starrocks.qe.ConnectContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PlannerProfile {
+    private ConnectContext ctx;
+
+    public static class ScopedTimer implements AutoCloseable {
```

Comment:
extend StopWatch maybe more easy？

Reply:
Is it? Because in theory I can do more than `StopWatch` in this class, liking recording how many times this function is called, and p99 latency of that function.  And `StopWatch` is wrapping of `System.currentTsInMillseconds` right? Is there any extra work it does?